.github/workflows/code.yml

```diff
@@ -0,0 +1,221 @@
+name: Code CI
+
+on:
+  push:
+    branches:
+      # Restricting to these branches and tags stops duplicate jobs on internal
+      # PRs but stops CI running on internal branches without a PR. Delete the
+      # next 5 lines to restore the original behaviour
+      - master
+      - main
+    tags:
+      - "*"
+  pull_request:
+  schedule:
+    # Run every Monday at 8am to check latest versions of dependencies
+    - cron: "0 8 * * MON"
+
+jobs:
+  lint:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: lint
+        run: |
+          pip install --user .[dev]
```

Comment:
Yes - because I'm using system python (which keeps the file terse) it fails to install . probably 
sudo pip install
would work but that is no prettier